datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1277,6 +1407,56 @@ pub(crate) fn from_substrait_literal(lit: &Literal) -> Result<ScalarValue> {
                 s,
             )
         }
+        Some(LiteralType::List(l)) => {
```

Comment:
I don't necessarily need these for this PR, so if it's preferrable I'm happy to split them into separate PRs - I just added them while coding to confirm that lists and structs inside lists are handled correctly.

Reply:
If it is convenient, using separate PRs would make the review process easier. Perhaps one PR could focus on the virtual table, while the other two could address the list and struct separately.

Reply:
Split those into https://github.com/apache/datafusion/pull/10615, https://github.com/apache/datafusion/pull/10622, https://github.com/apache/datafusion/pull/10640